src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtils.java

```diff
@@ -0,0 +1,61 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Contains utility methods for working with Xpath queries.
+ *
+ * @author Timur Tibeyev.
+ */
+public final class XpathUtils {
+    /** The map of {@link TokenTypes} to {@code String}. */
+    private static final Map<Integer, String> TOKEN_TYPES_TO_STRING_MAP = new HashMap<>();
+
+    /**
+     * Private utility constructor.
+     * @throws UnsupportedOperationException if called
+     */
+    private XpathUtils() {
+        throw new UnsupportedOperationException("do not instantiate.");
```

Comment:
I took `AnnotationUtility` class as sample, it has exception thrown inside constructor